src/table/virtualized-table/Body.tsx

```diff
@@ -52,44 +67,87 @@ const Body = (props: BodyProps) => {
     pageInfo,
   });
 
-  const itemData = useMemo(
-    () => ({ rowsInPage, columns, bodyStyle, isHoverEnabled }),
-    [rowsInPage, columns, bodyStyle, isHoverEnabled]
+  const itemData = useMemo<ItemData>(
+    () => ({ rowsInPage, columns, bodyStyle, isHoverEnabled, maxLineCount }),
+    [rowsInPage, columns, bodyStyle, isHoverEnabled, maxLineCount]
   );
 
+  useEffect(() => {
+    syncHeight(innerForwardRef.current?.clientHeight ?? 0, true);
+  }, [deferredRowCount, syncHeight, innerForwardRef]);
+
+  useEffect(() => {
+    syncHeight(innerForwardRef.current?.clientHeight ?? 0, false);
+
+    if (rowMeta.current.lastScrollToRatio === 1) {
+      // Hack to deal with the case when a user scrolls to the last row
+      // and data has not yet finished loading
+      gridRef.current?.scrollToItem({ rowIndex: deferredRowCount - 1, align: 'start' });
+    }
+  });
```

Comment:
It does run on every render and unfortunately it needs to do that because we want to stay on the last row when `lastScrollToRatio` is 1.

The `syncHeight` is her as an escape hatch to ensure that never end up out-of-sync when the other container is smaller than the inner container (react-window container).

Regarding "run twice". That should be because of `StrictMode`.

Reply:
I meant that you call `syncHeight` in  the upper useEffect with the same arguments as well. E.g. when `deferredRowCount` is updated, wouldn't `syncHeight` run from both useEffects?

Reply:
The arguments are different? The on that runs on every render will only "sync" given some condition is meet. Wheres the one running when `deferredRowCount` changes will always "sync" when called. But yes there is a case where but will "sync" with the same value. But I would be supprised if that caused two renders?

Reply:
I even double-checked that the arguments where the same, but I must need new glasses... This makes sense then :)